compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.cpp

```diff
@@ -129,7 +129,6 @@ void buildFlowTransformPassPipeline(OpPassManager &passManager,
       // Preprocess the input to a form more amenable for fusion
       .addPass(createRaiseSpecialOps)
       .addPass(createInterchangeGenericOpsPass)
-      .addPass(createCollapseDimsPass)
```

Comment:
What happens to the pass? Are you just testing?

Reply:
IIRC, we need to remove this pass because it technically is deprecated by [this pass](https://github.com/openxla/iree/blob/d354465a12182e59ed21918420b5b777b7a9b6db/compiler/src/iree/compiler/Dialect/Flow/Transforms/CollapseDimensions.cpp), as seen on this [PR](https://github.com/openxla/iree/pull/14656/files). 


However since 2d vector distribution is not supported yet , removing this pass caused some issues as seen on this [issue](https://github.com/openxla/iree/issues/15078).

But in order to do fusion of dequant and matvec without the ad-hoc `createFuseDequant` pass we need to get rid of this pass. Otherwise, the reduction dims of matvec would be collapsed before dispatch formation and won't allow us to fuse the dequant with it. 

I still need to check to see if N-d reduction distribution is working now though as seen on the issue mentioned above.

Reply:
OK, just checked, still broken, let me fix that up first, and move this to draft in the mean time. :)